Sources/ApolloCodegenLib/Templates/TemplateRenderer.swift

```diff
@@ -79,6 +80,10 @@ extension TemplateRenderer {
 
   private func renderSchemaFile(_ type: TemplateTarget.SchemaFileType) -> String {
     let namespace: String? = {
+      if case .cacheKeyResolutionExtension = type {
```

Comment:
Do none of the combinations below work for this file? At first look it seems like `case (true, nil):` should return the right string.

Reply:
By not having a namespace name, we don't call `wrappedinNamespace`. This needs an extension on `MySchemaAPI.Schema`. You can't do an extension on `Schema` inside of an extension on `MySchemaAPI`.

Though we could actually do the `case let (true, .some(schemaTypeNamespace)):` if we provided `Schema` as the `namespaceComponent`.

But then the template is just the static function, and it's **always** wrapped in the namespace. That's not really the intended use of this `TemplateRenderer`. It's to determine the wrapping context based on the config options.